### PR TITLE
Update plugin section - user guide

### PR DESCRIPTION
**Pull request by mhmohona**
## Explanation

Added plugin user guide for C, Rust and Go.

## What type of PR is this
/kind documentation

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by mhmohona** [comment]
@alabulei1 @q82419 I have completed working on this PR. Will you please have a look at it? 
Thank you!

File: docs/embed/use-case/plugin/go_sdk.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+sidebar_position: 3
+---
+
+## Using Plug-ins to Extend the Runtime in C
```
[the remark this answers is not in the document]
Comment (mhmohona) [reply docs/embed/use-case/plugin/go_sdk.md:5]: @alabulei1, yes the other steps are almost same but the code is different for each plugin. 

Here is the reference:
- For C - [API reference](https://wasmedge.org/docs/embed/c/reference/latest#plug-ins)
- For Go - [API reference](https://wasmedge.org/docs/embed/go/reference/latest#plug-ins)
- For Rust - [Plugin Manager doc](https://wasmedge.github.io/wasmedge-rust-sdk/wasmedge_sdk/plugin/struct.PluginManager.html)

**Review by q82419 — CHANGES_REQUESTED** [review]

File: docs/embed/use-case/plugin/go_sdk.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhmohona) [reply docs/embed/use-case/plugin/go_sdk.md:5]: ops, sorry. My bad. I forgot to change this.

File: docs/embed/use-case/plugin/c_sdk.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+/* Load the plug-ins in the default paths first. */
+WasmEdge_PluginLoadWithDefaultPaths();
+/* Create the configure context and add the WASI configuration. */
+WasmEdge_ConfigureContext *ConfCxt = WasmEdge_ConfigureCreate();
+WasmEdge_ConfigureAddHostRegistration(ConfCxt,
+                                        WasmEdge_HostRegistration_Wasi);
+WasmEdge_VMContext *VMCxt = WasmEdge_VMCreate(ConfCxt, NULL);
+WasmEdge_ConfigureDelete(ConfCxt);
+/*
+* The following API can retrieve the registered modules in the VM context,
```
Comment (q82419) [inline docs/embed/use-case/plugin/c_sdk.md:123]: Please actually lint the codes.

**Review by q82419 — CHANGES_REQUESTED** [review]

**Review by q82419 — APPROVED** [review]

**Comment by alabulei1** [comment]
I just released one question. Why this plugin guide located in embed section, not the extend section? @mhmohona

**Comment by mhmohona** [comment]
> I just released one question. Why this plugin guide located in the Embed section, not the extend section? @mhmohona

@alabulei1, I was instructed to add the user guide here. cc: @q82419

**Pull request merged**